src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -149,6 +163,10 @@ private void AnalyzeNode(
         if (objectType.Type == null || !objectType.Type.AllInterfaces.Contains(ienumerableType))
             return;
 
+        // Check if the diagnostic should be suppressed (e.g., inside CollectionBuilder methods)
```

Comment:
```suggestion
```

comment doesn't seem particularly useful given the method already has a doc comment iwth the same info